columnflow/plotting/plot_util.py

```diff
@@ -17,6 +17,51 @@
 od = maybe_import("order")
 
 
+def apply_variable_settings(hists: dict, variable_settings: dict | None = None) -> dict:
+    # check if there are variable settings to apply
+    if not variable_settings:
+        return hists
+    relevant_variables = set(list(hists.values())[0].axes.name) & set(variable_settings.keys())
+    if not relevant_variables:
+        return hists
+
+    # apply all settings
+    for var in relevant_variables:
+        var_settings = variable_settings[var]
+        for key, h in hists.items():
```

Comment:
Could you use `list(hists.items())` here? Just to be sure that no unwanted behavior occurs since you are overwriting the keys a few lines below while iterating (this is most probably ok in this instance since you are reusing the same keys, but it might be even a bit more expressive here with `list` as in "we are going to change the dict while iterating").